src/community/csw-iso/src/main/java/org/geoserver/csw/records/iso/FeatureCatalogueDescriptor.java

```diff
@@ -0,0 +1,143 @@
+
+    static {
+        FeatureTypeFactory typeFactory = new FeatureTypeFactoryImpl();
+
+        EmfComplexFeatureReader reader = EmfComplexFeatureReader.newInstance();
+
+        SchemaIndex index = null;
+        try {
+            index =
+                    reader.parse(
+                            new URL("http://schemas.opengis.net/iso/19139/20070417/gfc/gfc.xsd"));
+        } catch (IOException e) {
+            // this is fatal
+            throw new RuntimeException(e);
+        }
+
+        FeatureTypeRegistry featureTypeRegistry =
+                new FeatureTypeRegistry(
+                        MetaDataDescriptor.NAMESPACES,
+                        typeFactory,
+                        new RecordFeatureTypeRegistryConfiguration("FC_FeatureCatalogue_Type"),
+                        true);
+
+        featureTypeRegistry.addSchemas(index);
+
+        FEATURECATALOGUE_TYPE =
+                (FeatureType)
+                        featureTypeRegistry.getAttributeType(
+                                new NameImpl(
+                                        MetaDataDescriptor.NAMESPACE_GFC,
+                                        "FC_FeatureCatalogue_Type"));
+        FEATURECATALOGUE_DESCRIPTOR =
+                featureTypeRegistry.getDescriptor(
+                        new NameImpl(MetaDataDescriptor.NAMESPACE_GFC, "FC_FeatureCatalogue"),
+                        null);
+    }
+
+    @Override
+    public String getOutputSchema() {
+        return MetaDataDescriptor.NAMESPACE_GMD;
+    }
+
+    @Override
+    public List<Name> getPropertiesForElementSet(ElementSetType elementSet) {
+        switch (elementSet) {
+            case BRIEF:
+                return Collections.emptyList();
+            case SUMMARY:
+                return Collections.emptyList();
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public NamespaceSupport getNamespaceSupport() {
+        return MetaDataDescriptor.NAMESPACES;
+    }
+
+    @Override
+    public Query adaptQuery(Query query) {
+
+        return query;
+    }
+
+    @Override
+    public AttributeDescriptor getFeatureDescriptor() {
+        return FEATURECATALOGUE_DESCRIPTOR;
+    }
+
+    @Override
+    public String getBoundingBoxPropertyName() {
+        return null;
+    }
+
+    @Override
+    public List<Name> getQueryables() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public String getQueryablesDescription() {
+        return null;
+    }
+
+    @Override
+    public PropertyName translateProperty(Name name) {
+        return null;
+    }
+
+    @Override
+    public void verifySpatialFilters(Filter filter) {}
+
+    // singleton
+
+    private FeatureCatalogueDescriptor() {}
+
+    private static FeatureCatalogueDescriptor INSTANCE;
+
+    public static FeatureCatalogueDescriptor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = GeoServerExtensions.bean(FeatureCatalogueDescriptor.class);
```

Comment:
This mix between spring and singletons instantiated manually may end in situations where the bean is not return because it doesn't exists but because it was not instantiated yet.